### DISCUSSION TITLE
How to get icons or images from URLs (instead of local files)?

### DISCUSSION THREAD

**Discussion by jamesbraza** [nunocoracao/blowfish#888] [Q&A]
I am trying to avoid copying a bunch of images to my website's file system for things like:

- Icons (e.g. past university) into `assets/icons/`
- Article headers into `featured.png`

Remote URLs is already possible for `figure` shortcode because `src` also supports URLs.

Are there any ways to achieve this?

**Comment by wolfspyre** [comment]
yeah. this is something I think would be useful as well... 
MY (perhaps naive)  thought here was to introduce page-scoped frontmatter knobs for (feature,background) img... and have that support urls... however... this would likely come with the limitations that any resource not within a pagebundle is subject to 
(*I don't think that hugo can/will elegantly do image processing/resizing on external assets... but wth do I know?*)

However....
What's the expected end result in your mind? 
Would you expect hugo to collect those external assets at build-time and put static copies of them in place for the static output?

Would you expect hugo to just include img src tags? cause that can be problematic for pageload times... especially for latent assets used as thematic elements... 

now, if you're simply saying 'I'd like to only have ONE copy of _insert-image-name-here_ in my amalgamated static site blob...  rather than a copy of it in every page bundle I want a background for....  I can completely get behind this preference.

**Reply by jamesbraza** [reply]
I am down with whatever, I think you have some good ideas!  I like the idea of making them static at build-time too

**Comment by nunocoracao** [comment]
Hello @jamesbraza and @wolfspyre - is this reflected in a ticket already?
If not, would one of you mind creating that with all the info? 

From the top of my head - I think there are several requests for this feature but I am not sure it's within the theme scope VS Hugo's scope

**Comment by wolfspyre** [comment]
I'd think the latter option would be theme-scoped: 
> I'd like to be able to have ONE copy of insert-image-name-here in my amalgamated static site blob... rather than a copy of it in every page bundle I want a background for.... 

I'd think that would essentially just be a mechanism to use a file stored in `/assets/` or `/static/` as the featured/background/thumb img source... as opposed to a pagebundle resource...

Where this would also be additionally useful is for normal md files within a pagebundle that one would want a background on... as currently that's not possible unless I'm mistaken?

$.02 @nunocoracao ?

**Comment by fguisso** [comment]
Any updates on this? I need something to use the same URL for posts that don't have a featured image, this will avoid duplicating similar images for posts without thumbn.
